src/server/req_register.c

```diff
@@ -675,6 +857,121 @@ depend_on_exec(job *pjob)
+ *
+ * @param[in]	pjob	-	job
+ *
+ * @return	error code
+ * @retval	0	: success
+ */
+int depend_runone_remove_dependency(job *pjob)
+{
+	struct depend     *pdep;
+	struct depend_job *pdj;
+	struct job	  *d_pjob;
+
+	if (pjob == NULL)
+		return (0);
+
+	pdep = find_depend(JOB_DEPEND_TYPE_RUNONE, &pjob->ji_wattr[(int)JOB_ATR_depend]);
+	if (pdep) {
+		for (pdj = (struct depend_job *)GET_NEXT(pdep->dp_jobs);
+		     pdj != NULL; pdj = (struct depend_job *)GET_NEXT(pdj->dc_link)) {
+			d_pjob = find_job(pdj->dc_child);
+			if (d_pjob) {
+				struct depend_job *temp_pdj = NULL;
+				temp_pdj = find_dependjob(find_depend(JOB_DEPEND_TYPE_RUNONE, &d_pjob->ji_wattr[(int)JOB_ATR_depend]),
+					                  pjob->ji_qs.ji_jobid);
+				if (temp_pdj) {
+				        del_depend_job(temp_pdj);
+					d_pjob->ji_wattr[(int)JOB_ATR_depend].at_flags |= ATR_VFLAG_MODIFY|ATR_VFLAG_MODCACHE;
+				}
+			}
+		}
+	}
+	return (0);
+}
+
+/**
+ * @brief
+ * 		Helper function that goes through all dependent jobs with runone dependency
+ *		on the given job and puts all them on system hold.
+ * @note
+ *		This function is called from req_strtjob2().
+ *
+ * @param[in]	pjob	-	job
+ *
+ * @return	error code
+ * @retval	0	: success
+ */
+int depend_runone_hold_all(job *pjob)
+{
+	struct depend     *pdep;
+	struct depend_job *pdj;
+	struct job	  *d_pjob;
+
+	if (pjob == NULL)
+		return (0);
+
+	pdep = find_depend(JOB_DEPEND_TYPE_RUNONE, &pjob->ji_wattr[(int)JOB_ATR_depend]);
+	if (pdep) {
+		for (pdj = (struct depend_job *)GET_NEXT(pdep->dp_jobs);
+		     pdj != NULL; pdj = (struct depend_job *)GET_NEXT(pdj->dc_link)) {
+			d_pjob = find_job(pdj->dc_child);
+			if (d_pjob) {
+				d_pjob->ji_wattr[(int)JOB_ATR_hold].at_val.at_long |= HOLD_s;
+		                d_pjob->ji_wattr[(int)JOB_ATR_hold].at_flags |= ATR_VFLAG_SET|ATR_VFLAG_MODCACHE;
+				(void)svr_setjobstate(d_pjob, JOB_STATE_HELD, JOB_SUBSTATE_HELD);
+			}
+		}
+	}
+	return (0);
+}
+
+/**
+ * @brief
+ * 		Helper function that goes through all dependent jobs with runone dependency
+ *		on the given job and releases them of system hold.
+ * @note
+ *		This function is called from req_rerunjob2().
+ *
+ * @param[in]	pjob	-	job
+ *
+ * @return	error code
+ * @retval	0	: success
+ */
+int depend_runone_release_all(job *pjob)
+{
+	struct depend     *pdep;
+	struct depend_job *pdj;
+	struct job	  *d_pjob;
+	int	newstate;
+	int	newsub;
+
+	if (pjob == NULL)
+		return (0);
+
+	pdep = find_depend(JOB_DEPEND_TYPE_RUNONE, &pjob->ji_wattr[(int)JOB_ATR_depend]);
+	if (pdep) {
+		pdj = (struct depend_job *)GET_NEXT(pdep->dp_jobs);
+		for (pdj = (struct depend_job *)GET_NEXT(pdep->dp_jobs);
+		     pdj != NULL; pdj = (struct depend_job *)GET_NEXT(pdj->dc_link)) {
+			d_pjob = find_job(pdj->dc_child);
+			if (d_pjob) {
+				d_pjob->ji_wattr[(int)JOB_ATR_hold].at_val.at_long &= ~HOLD_s;
+		                d_pjob->ji_wattr[(int)JOB_ATR_hold].at_flags |= ATR_VFLAG_SET|ATR_VFLAG_MODCACHE;
```

Comment:
Does this become set, even though we're removing a hold? Also, what about MODIFY? And fix the spacing

Reply:
Good catch!
You are right, this shouldn't be set at all.